src/Compilers/CSharp/Test/Syntax/Syntax/SyntaxTests.cs

```diff
@@ -276,5 +278,58 @@ public void IsAttributeTargetSpecifier()
                 }
             }
         }
+
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/72300")]
+        public void TestAllKindsReturnedFromGetKindsMethodsExist()
+        {
+            foreach (var method in typeof(SyntaxFacts).GetMethods(BindingFlags.Public | BindingFlags.Static))
+            {
+                if (method.ReturnType == typeof(IEnumerable<SyntaxKind>) && method.GetParameters() is [])
+                {
+                    foreach (var kind in (IEnumerable<SyntaxKind>)method.Invoke(null, null))
+                    {
+                        Assert.True(Enum.IsDefined(typeof(SyntaxKind), kind), $"Nonexistent kind '{kind}' returned from method '{method.Name}'");
+                    }
+                }
+            }
+        }
+
+        [Theory, WorkItem("https://github.com/dotnet/roslyn/issues/72300")]
+        [InlineData(nameof(SyntaxFacts.GetContextualKeywordKinds), SyntaxKind.YieldKeyword, SyntaxKind.FileKeyword)]
+        [InlineData(nameof(SyntaxFacts.GetPunctuationKinds), SyntaxKind.TildeToken, SyntaxKind.GreaterThanGreaterThanGreaterThanEqualsToken)]
+        [InlineData(nameof(SyntaxFacts.GetReservedKeywordKinds), SyntaxKind.BoolKeyword, SyntaxKind.ImplicitKeyword)]
+        public void TestRangeBasedGetKindsMethodsReturnExpectedResults(string methodName, SyntaxKind firstExpectedKind, SyntaxKind lastExpectedKind)
+        {
+            var method = typeof(SyntaxFacts).GetMethod(methodName, BindingFlags.Public | BindingFlags.Static);
+
+            Assert.NotNull(method);
+            Assert.Equal(0, method.GetParameters().Length);
+            Assert.Equal(typeof(IEnumerable<SyntaxKind>), method.ReturnType);
+
+            foreach (var kind in (IEnumerable<SyntaxKind>)method.Invoke(null, null))
+            {
+                Assert.InRange(kind, firstExpectedKind, lastExpectedKind);
+            }
+        }
+
+        [Fact]
+        public void TestGetPreprocessorKeywordKindsReturnsExpectedResults()
```

Comment:
>TestGetPreprocessorKeywordKindsReturnsExpectedResults [](http://example.com/codeflow?start=20&length=53)

It looks like this test is also failing to fulfil the requested aspect - "a test that would fail when ```GetPreprocessorKeywordKinds``` fails to return all expected kinds"